include/npcomp/Dialect/ATen/ATen.td

```diff
@@ -0,0 +1,161 @@
+//===- ATen.td ---------------------------------------------*- tablegen -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifdef OP_BASE
+#else
+include "mlir/IR/OpBase.td"
+#endif // OP_BASE
+
+#ifdef ATEN_OPS
+#else
+#define ATEN_OPS
+#endif
+
+include "mlir/Interfaces/SideEffectInterfaces.td"
+include "npcomp/Dialect/ATen/ATenOpInterface.td"
+
+def aten_Dialect : Dialect {
+  let name = "aten";
+  let cppNamespace = "aten";	
+}
+
+class aten_Op<string mnemonic, list<OpTrait> traits = [StatisticsOpInterface]> :
+    Op<aten_Dialect, mnemonic, traits>;
+
+
+// Most ops are automatically generated from pytorch specs.
+include "npcomp/Dialect/ATen/ATenOps.td"
+
+
+def aten_BatchNormOp: aten_Op<"batch_norm", [NoSideEffect, StatisticsOpInterface]>,
+                      Results<(outs AnyTensor:$output, AnyTensor:$save_mean, AnyTensor:$save_invstd)> {
+  let arguments = (
+    ins AnyType:$arg0,
```

Comment:
Any chance we could get actual symbolic argument names? Esp for BatchNorm which has a lot of parameters.

Reply:
There are a few Ops in this file that are hand-written and should go away, I think.  Most of the ops are automatically generated in ATenOps.td (with names derived from pytorch)

Reply:
Moved to issue